odml/tools/rdf_converter.py

```diff
@@ -0,0 +1,141 @@
+import sys
+import uuid
+
+from rdflib import Graph, BNode, Literal, URIRef
+from rdflib.namespace import XSD, RDF
+
+import odml
+import odml.format as format
+
+try:
+    unicode = unicode
+except NameError:
+    unicode = str
+
+odmlns = format.Format.namespace()
+
+
+class RDFWriter:
+    """ 
+    Creates the RDF graph storing information about the odML document 
+    """
+
+    def __init__(self, odml_documents, hub_id=None):
+        self.docs = odml_documents
+        self.hub_id = hub_id
+        self.hub_root = None
+        self.g = Graph()
+        self.g.bind("odml", odmlns)
+
+    def create_hub_root(self):
+        if self.hub_root is None:
+            if self.hub_id is None:
+                self.hub_root = URIRef(odmlns + str(uuid.uuid4()))
+            else:
+                self.hub_root = URIRef(odmlns + self.hub_id)
+
+    def convert_to_rdf(self, docs):
+        self.create_hub_root()
+        if docs:
+            self.g.add((self.hub_root, odmlns.Hub, URIRef(odmlns.Hub)))
+            for doc in docs:
+                self.save_element(doc)
+        return self.g
+
+    def save_element(self, e, node=None):
+        """
+        Save the current element to the RDF graph
+        :param e: current element 
+        :param node: Blank node to pass the earlier created node to inner elements
+        :return: the RDF graph 
+        """
+        fmt = e._format
+
+        if not node:
+            curr_node = URIRef(odmlns + str(e.id))
+        else:
+            curr_node = node
+
+        self.g.add((curr_node, fmt.rdf_type(), URIRef(fmt.rdf_type())))
```

Comment:
To properly add the ontology type of the class use `self.g.add((curr_node, RDF.type, URIRef(fmt.rdf_type())))`.